finagle-core/src/main/scala/com/twitter/finagle/service/StatsFilter.scala

```diff
@@ -158,27 +159,34 @@ class StatsFilter[Req, Rep](
     val elapsed = Stopwatch.start()
 
     outstandingRequestCount.increment()
-    service(request).respond { response =>
-      outstandingRequestCount.decrement()
-      if (!isBlackholeResponse(response)) {
-        dispatchCount.incr()
-        responseClassifier.applyOrElse(
-          ReqRep(request, response),
-          ResponseClassifier.Default
-        ) match {
-          case ResponseClass.Failed(_) =>
-            latencyStat.add(elapsed().inUnit(timeUnit))
-            response match {
-              case Throw(e) =>
-                exceptionStatsHandler.record(statsReceiver, e)
-              case _ =>
-                exceptionStatsHandler.record(statsReceiver, SyntheticException)
-            }
-          case ResponseClass.Successful(_) =>
-            successCount.incr()
-            latencyStat.add(elapsed().inUnit(timeUnit))
+
+    try {
```

Comment:
let's scope the try/catch block more narrowly.  I think we want to be able to reuse the behavior from the rest of the method, so how about we convert a `service(request)` that throws an exception into one that returns a failed future?  we could make this instead be

```scala
val result = try {
  service(request)
}  catch { case NonFatal(e) => 
  Future.exception(e)
}

result.respond { ...
```

what do you think?